.evergreen/tasks.in.yml

```diff
@@ -135,17 +135,57 @@ tasks:
       - func: apply-compass-target-expansion
         vars:
           compass_distribution: compass
-      - func: get-packaged-app
+      - func: test-web-sandbox
         vars:
-          compass_distribution: compass
-      - func: test-packaged-app
-        vars: <% for (const [key, value] of Object.entries(testPackagedTask.vars)) { %>
-          <% out(`${key}: '${value}'`) } %>
+          mongodb_version: 'latest-alpha-enterprise'
+          browser_name: 'firefox'
           compass_distribution: compass
           debug: 'compass-e2e-tests*,electron*,hadron*,mongo*'
+
+<% for (const distribution of ["compass", "compass-isolated", "compass-readonly"]) { %>
+  - name: package-<%= distribution %>
+    tags: ['required-for-publish', 'run-on-pr']
+    commands:
+      - func: prepare
+      - func: install
+      - func: bootstrap
+        vars:
+          scope: '@mongodb-js/webpack-config-compass'
+      - func: apply-compass-target-expansion
+        vars:
+          compass_distribution: <%= distribution %>
+      - func: spawn-signing-server
+      - func: package
+        vars:
+          debug: 'hadron*,mongo*,compass*,electron*,sign*'
+          compass_distribution: <%= distribution %>
+      - func: verify-artifacts
+      - func: save-all-artifacts
+        vars:
+          compass_distribution: <%= distribution %>
 <% } %>
-<% for (const testPackagedTask of tasks['test-web-sandbox']) { %>
-  - name: <% out(testPackagedTask.name) %>
+
+<%
+const serverVersions = [
+  { name: '40x-community', version: '4.0.x' },
+  { name: '40x-enterprise', version: '4.0.x-enterprise' },
+  { name: '42x-community', version: '4.2.x' },
+  { name: '42x-enterprise', version: '4.2.x-enterprise' },
+  { name: '44x-community', version: '4.4.x' },
+  { name: '44x-enterprise', version: '4.4.x-enterprise' },
+  { name: '5x-community', version: '5.x.x' },
+  { name: '5x-enterprise', version: '5.x.x-enterprise' },
+  { name: '60x-community', version: '6.0.x' },
+  { name: '60x-enterprise', version: '6.0.x-enterprise' },
+  { name: '70x-community', version: '7.0.x' },
+  { name: '70x-enterprise', version: '7.0.x-enterprise' },
+  { name: '80x-community', version: '>= 8.0.0-rc4' },
+  { name: '80x-enterprise', version: '>= 8.0.0-rc4-enterprise' },
+  { name: 'latest-alpha', version: 'latest-alpha-enterprise' }
+];
```

Comment:
Any way we can share this with the buildvariants? Feels like it would be easy to mess up if you need to remember that adding anything here should also require a manual change in the variants. Maybe we can at least write a comment about this?

Reply:
we can definitely share them yeah, also evergreen would complain here if you mess something up though. The name of the tasks must match, and since all of them are required for publish, each task must appear in at least one build variant.

Reply:
Hmmm, are you sure evergreen will catch this? I removed some test-packaged records from the buildvariant and ran evergreen validate, but it didn't fail:

```diff
--- a/.evergreen/buildvariants.yml
+++ b/.evergreen/buildvariants.yml
@@ -87,8 +87,6 @@ buildvariants:
       - name: package-compass
         variant: test-and-package-ubuntu
     tasks:
-      - name: test-packaged-app-40x-community
-      - name: test-packaged-app-40x-enterprise
       - name: test-packaged-app-42x-community
       - name: test-packaged-app-42x-enterprise
       - name: test-packaged-app-44x-community
```

![image](https://github.com/mongodb-js/compass/assets/5036933/ef7daa50-88d5-4dcc-8e13-3e7447fd60d4)

Maybe to clarify, the potentially error-prone scenario I'm imagining: you add a new variant in the tasks array, it generates the task, but it's not obvious that the task needs to be added to the buildvariant too manually

Reply:
Ah, sorry, I read your message incorrectly, at least one! Okay, so I do see the warning now, but it still allows me to schedule the patch in evergreen without failures with removed task just missing from the required list there. Maybe we can add `evergreen verify` to pre-commit and make it fail on warnings somehow?

Reply:
ended up refactoring it anyway :)

Reply:
Maybe we can add evergreen verify to pre-commit and make it fail on warnings somehow

what about adding it to the `template.yml.js` i end up running it all of the times anyway